tests/common.py

```diff
@@ -379,11 +379,36 @@ def async_fire_mqtt_message(hass, topic, payload, qos=0, retain=False):
 fire_mqtt_message = threadsafe_callback_factory(async_fire_mqtt_message)
 
 
+@ha.callback
+def async_fire_time_changed_exact(
+    hass: HomeAssistant, datetime_: datetime | None = None, fire_all: bool = False
+) -> None:
+    """Fire a time changed event at an exact microsecond.
+
+    Consider that its not possible to actually achieve an exact microsecond
```

Comment:
it is not possible

Reply:
https://github.com/home-assistant/core/pull/82349